### ISSUE TITLE
Cannot read property 'typeName' of undefined

### ISSUE THREAD

**Issue by johnmcguin** [NodeSquarespace/node-squarespace-server#133]
Hi @kitajchuk, I recently tried to switch a project over to start developing locally, which I've done before. This time I encountered this error. I recently updated to the newest version (0.4.3) and tried running the sqs buster, but keep getting this error. The page will load at first but will then error out immediately after load. Any thoughts appreciated. Thanks!

/usr/local/lib/node_modules/node-squarespace-server/squarespace-template.js:886
    typeName = pageJson.collection.typeName;
                                  ^
TypeError: Cannot read property 'typeName' of undefined
    at getTemplateKey (/usr/local/lib/node_modules/node-squarespace-server/squarespace-template.js:886:35)
    at Object.renderTemplate (/usr/local/lib/node_modules/node-squarespace-server/squarespace-template.js:443:23)
    at /usr/local/lib/node_modules/node-squarespace-server/squarespace-server.js:228:29
    at /usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-middleware/squarespace-middleware.js:311:13
    at Request._callback (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-middleware/squarespace-middleware.js:234:9)
    at Request.self.callback (/usr/local/lib/node_modules/node-squarespace-server/node_modules/request/request.js:198:22)
    at Request.emit (events.js:110:17)
    at Request.<anonymous> (/usr/local/lib/node_modules/node-squarespace-server/node_modules/request/request.js:1063:14)
    at Request.emit (events.js:129:20)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/node-squarespace-server/node_modules/request/request.js:1009:12)
Johns-MacBook-Pro-3:un-sdgs-site johnmcguin$ sqs --version
> sqs-log Node Squarespace Server version 0.4.3

**Comment by kitajchuk** [comment]
What page is throwing this error for you? What type of object is it? A collection, item. What type is it?

**Comment by mattnull** [comment]
I've seen this error anytime I use a template other than base-template. While loading any page. I think this is related to https://github.com/NodeSquarespace/node-squarespace-server/issues/126

**Comment by kitajchuk** [comment]
It seems unlikely that this would be related to `browser-sync`. I'm having no luck reproducing it currently. Which template are you using?

**Comment by kitajchuk** [comment]
@johnmcguin - Check my first comment. It would help if you can paste the collection object from the page you are trying to request as well as your layouts information from your template.conf file.

**Comment by mattnull** [comment]
@kitajchuk I think I am noticing a pattern now. It seems like sqs is attempting to download a font asset as an html file in this case that I am seeing. You can see in the screenshot below that its adding a .html extension to the font file. In another scenerio I saw the same thing happen wtih an svg image, and another case that showed sqs fetching page-undefined.html.
https://www.dropbox.com/s/ltt240zq13pgayl/Screenshot%202015-09-28%2014.33.54.png?dl=0

**Comment by kitajchuk** [comment]
BAM! That looks legit to me. How are you linking to those assets in your template?

**Comment by mwickett** [comment]
I'm seeing a similar issue (after updating after #134 )
![screenshot 2015-09-28 16 42 53](https://cloud.githubusercontent.com/assets/845983/10148178/14c7eafa-6600-11e5-9f41-b00466286626.png)

**Comment by mattnull** [comment]
@kitajchuk we are including them in a css file. The css file references the font file via @font-face. In template.conf our fontawesome.css is in the "stylesheets" array.

**Comment by kitajchuk** [comment]
If you link to `/assets/fonts/fontawesome-webfont.woff2` instead does that work? I'm assuming you have a `fonts` directory in your assets folder, yeah?

**Comment by mattnull** [comment]
@kitajchuk They are in the assets folder. That fixed my issue.

**Comment by kitajchuk** [comment]
Awesome!

**Comment by mattnull** [comment]
Thanks @kitajchuk

**Comment by skrichten** [comment]
I just started getting this same error but in a different use case.  I am using the Wells template and I have a folder in the main navigation.  In the folder I have a "page".  I want the user to get the first page in the folder when they click on it so in my nav template I have...

'''
<a href="{fullUrl}">{navigationTitle}</a>
'''

... even if the item is a folder. This works on the actual squarespace site, but I get the "Cannot read property 'typeName' of undefined"  when I click on the folder in the navigation.

**Comment by kitajchuk** [comment]
Hmm. @skrichten - Can you paste the collection object JSON for the page you would like to get working?

**Comment by skrichten** [comment]
Right now, its just the contact page that comes with the template, and I put it in a folder.

```
{
  "mainContent" : "<div class=\"sqs-layout sqs-grid-12 columns-12\" data-type=\"page\" data-updated-on=\"1440698878092\" id=\"page-5603fb6ae4b0777b299c4762\"><div class=\"row sqs-row\"><div class=\"col sqs-col-12 span-12\"><div class=\"sqs-block html-block sqs-block-html\" data-block-type=\"2\" id=\"block-9914d46e4cd9dd613009\"><div class=\"sqs-block-content\"><h1>World Renowned Photographer. Will travel.</h1></div></div><div class=\"sqs-block image-block sqs-block-image\" data-block-type=\"5\" id=\"block-9a64f2277e4c8497edfa\"><div class=\"sqs-block-content\">\n\t<div class=\"image-block-outer-wrapper layout-caption-below \">\n\t\n\t\t<div class=\"intrinsic\" style=\"max-width:100%;\">\n\t\t\t\n\t\t\t\t<div style=\"padding-bottom:66.5671615600586%;\" class=\"image-block-wrapper   \" data-description=\"\">\n\t\t\t\t\t<noscript><img src=\"http://static1.squarespace.com/static/551d81c0e4b0c387e6e080df/t/5603fb6ae4b0777b299c4763/1344631508108/download.jpeg\"  alt=\"Transient\"  /></noscript><img class=\"thumb-image\" alt=\"Transient\" data-src=\"http://static1.squarespace.com/static/551d81c0e4b0c387e6e080df/t/5603fb6ae4b0777b299c4763/1344631508108/download.jpeg\" data-image=\"http://static1.squarespace.com/static/551d81c0e4b0c387e6e080df/t/5603fb6ae4b0777b299c4763/1344631508108/download.jpeg\" data-image-dimensions=\"670x446\" data-image-focal-point=\"0.5,0.5\" data-load=\"false\" data-image-id=\"5603fb6ae4b0777b299c4763\" data-type=\"image\" />\n\t\t\t\t</div>\n\t\t\t\n\n\t\t\t\n\n\t\t</div>\n\t\n\t</div>\n</div></div><div class=\"row sqs-row\"><div class=\"col sqs-col-6 span-6\"><div class=\"sqs-block form-block sqs-block-form\" data-block-type=\"9\" id=\"block-dd2e58d7a938ab08c767\"><div class=\"sqs-block-content\">\n\n<div class=\"form-wrapper\">\n\n  \n\n  \n\n    <div class=\"form-inner-wrapper\">\n\n      <form autocomplete=\"on\" action=\"https://steve-krichten.squarespace.com\" method=\"POST\" onsubmit=\"return (function(form) { Y.use\n      ('squarespace-form-submit', 'node',\n      function (Y) { (new Y.Squarespace.FormSubmit({ formNode: Y.Node(form) })).submit('5603fb6ae4b0777b299c4764', '5603fb6ae4b0777b299c4762') }); return false;})(this)\">\n\n        \n\n          <div class=\"field-list clear\">\n\n          \n\n              \n\n              \n\n              \n                <div id=\"text-11\" class=\"form-item field text required\">\n                  <label class=\"title\" for=\"text-11-field\">Name <span class=\"required\">*</span></label>\n                  \n                  <input class=\"field-element text\" type=\"text\" id=\"text-11-field\" />\n                </div>\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n          \n\n              \n\n              \n\n              \n\n              \n\n              \n                <div id=\"email2\" class=\"form-item field email required\">\n                  <label class=\"title\" for=\"email2-field\">Email Address <span class=\"required\">*</span></label>\n                  \n                  <input class=\"field-element\" name=\"email\" x-autocompletetype=\"email\" type=\"text\" spellcheck=\"false\" id=\"email2-field\" />\n                </div>\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n          \n\n              \n\n              \n\n              \n                <div id=\"text3\" class=\"form-item field text required\">\n                  <label class=\"title\" for=\"text3-field\">Subject <span class=\"required\">*</span></label>\n                  \n                  <input class=\"field-element text\" type=\"text\" id=\"text3-field\" />\n                </div>\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n          \n\n              \n\n              \n\n              \n\n              \n                <div id=\"textarea4\" class=\"form-item field textarea\">\n                  <label class=\"title\" for=\"textarea4-field\">Message</label>\n                  \n                  <textarea class=\"field-element \" id=\"textarea4-field\" ></textarea>\n                </div>\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n              \n\n          \n\n          </div>\n\n        \n\n        \n        <div class=\"form-button-wrapper\">\n          <input class=\"button sqs-system-button sqs-editable-button\" type=\"submit\" value=\"Send\"/>\n        </div>\n        \n\n        <div class=\"hidden form-submission-text\"><p>Thanks for your message!</p></div>\n\n        <div class=\"hidden form-submission-html\" data-submission-html=\"\"></div>\n      </form>\n\n    </div>\n\n  \n\n</div>\n</div></div></div><div class=\"col sqs-col-6 span-6\"><div class=\"sqs-block html-block sqs-block-html\" data-block-type=\"2\" id=\"block-eec63e0701c7008c8ca3\"><div class=\"sqs-block-content\"><p>Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Integer posuere erat a ante venenatis dapibus posuere velit aliquet blandit tempus porttitor. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Lorem ipsum dolor sit amet, consectetur adipiscing.</p><blockquote>Vestibulum id ligula porta felis euismod semper.</blockquote><p>Etiam porta sem malesuada magna mollis euismod. Donec id elit non mi porta gravida at eget metus. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Maecenas sed diam eget risus varius blandit sit amet non magna. Maecenas sed diam eget risus varius blandit sit amet non magna. Cras justo odio, dapibus ac facilisis in, egestas eget quam.</p><p><a href=\"//twitter.com\">Twitter</a> / <a href=\"//facebook.com\">Facebook</a> / <a href=\"mailto:bestphotographer@email.com\">bestphotographer@email.com</a></p></div></div></div></div></div></div></div>",
  "website" : {
    "id" : "551d81c0e4b0c387e6e080df",
    "identifier" : "steve-krichten",
    "contentModifiedOn" : 1443622289069,
    "cloneable" : false,
    "developerMode" : true,
    "siteStatus" : { },
    "language" : "en-US",
    "timeZone" : "America/New_York",
    "machineTimeZoneOffset" : -14400000,
    "timeZoneOffset" : -14400000,
    "timeZoneAbbr" : "EDT",
    "siteTitle" : "Wells Test",
    "siteDescription" : "",
    "shareButtonOptions" : {
      "7" : true,
      "3" : true,
      "5" : true,
      "4" : true,
      "6" : true,
      "2" : true,
      "1" : true,
      "8" : true
    },
    "authenticUrl" : "http://steve-krichten.squarespace.com",
    "baseUrl" : "http://steve-krichten.squarespace.com",
    "typekitId" : "",
    "statsMigrated" : false,
    "imageMetadataProcessingEnabled" : false
  },
  "websiteSettings" : {
    "id" : "551d81c0e4b0c387e6e080e2",
    "websiteId" : "551d81c0e4b0c387e6e080df",
    "type" : "Non-Profit",
    "subjects" : [ ],
    "simpleLikingEnabled" : true,
    "mobileInfoBarSettings" : {
      "isContactEmailEnabled" : false,
      "isContactPhoneNumberEnabled" : false,
      "isLocationEnabled" : false,
      "isBusinessHoursEnabled" : false
    },
    "lastAgreedTermsOfService" : 2,
    "defaultPostFormat" : "%y/%m/%d/%t",
    "commentLikesAllowed" : true,
    "commentAnonAllowed" : true,
    "commentThreaded" : true,
    "commentApprovalRequired" : false,
    "commentAvatarsOn" : true,
    "commentSortType" : 2,
    "commentFlagThreshold" : 0,
    "commentFlagsAllowed" : true,
    "commentEnableByDefault" : true,
    "disqusShortname" : "",
    "homepageTitleFormat" : "%s",
    "collectionTitleFormat" : "%c \u2014 %s",
    "itemTitleFormat" : "%i \u2014 %s",
    "commentsEnabled" : true,
    "contactPhoneNumber" : "",
    "allowSquarespacePromotion" : true,
    "storeSettings" : {
      "returnPolicy" : null,
      "termsOfService" : null,
      "privacyPolicy" : null,
      "orderRefundedSubjectFormat" : "%s: Order #%o Refunded",
      "orderRefundedTitleFormat" : "Order Refunded: #%o",
      "paymentSettings" : { },
      "expressCheckout" : false,
      "useLightCart" : false,
      "showNoteField" : false,
      "currenciesSupported" : [ "AUD", "CHF", "EUR", "GBP", "USD", "CAD" ],
      "defaultCurrency" : "USD",
      "selectedCurrency" : "USD",
      "measurementStandard" : 1,
      "showCustomCheckoutForm" : false,
      "enableMailingListOptInByDefault" : true,
      "businessName" : "HerizonTest",
      "businessId" : "",
      "stripeConnected" : false,
      "isLive" : false,
      "storeState" : 3
    },
    "useEscapeKeyToLogin" : true,
    "ssBadgeType" : 1,
    "ssBadgePosition" : 4,
    "ssBadgeVisibility" : 1,
    "ssBadgeDevices" : 1
  },
  "collection" : {
    "id" : "5603fb6ae4b0777b299c4762",
    "websiteId" : "551d81c0e4b0c387e6e080df",
    "enabled" : true,
    "starred" : false,
    "type" : 10,
    "ordering" : 2,
    "title" : "Contact",
    "navigationTitle" : "Contact",
    "urlId" : "contact",
    "itemCount" : 0,
    "updatedOn" : 1440698878095,
    "description" : "",
    "publicCommentCount" : 0,
    "folder" : false,
    "dropdown" : false,
    "tags" : [ ],
    "categories" : [ ],
    "homepage" : false,
    "typeName" : "page",
    "synchronizing" : false,
    "fullUrl" : "/contact/",
    "passwordProtected" : false,
    "typeLabel" : "page"
  },
  "authenticatedAccount" : {
    "id" : "551d81bee4b07556d7cabca5",
    "lastLoginOn" : 1443643290480,
    "lastActiveOn" : 1443643290480,
    "isDeactivated" : false,
    "deleted" : false,
    "displayName" : "Steve Krichten",
    "firstName" : "Steve",
    "lastName" : "Krichten",
    "emailVerified" : false,
    "websiteUrl" : "",
    "bio" : "",
    "location" : {
      "mapLat" : 40.7207559,
      "mapLng" : -74.00076130000002,
      "mapZoom" : 12,
      "markerLat" : 40.7207559,
      "markerLng" : -74.00076130000002
    },
    "revalidateTimestamp" : 1427997118807,
    "systemGenerated" : false
  },
  "template" : {
    "mobileStylesEnabled" : true
  },
  "shoppingCart" : {
    "websiteId" : "551d81c0e4b0c387e6e080df",
    "isPurchased" : false,
    "created" : 1443648655266,
    "entries" : [ ],
    "shippingOptions" : [ ],
    "shippingLocation" : { },
    "taxLineItems" : [ ],
    "coupons" : [ ],
    "validCoupons" : [ ],
    "invalidCoupons" : [ ],
    "subtotalCents" : 0,
    "taxCents" : 0,
    "shippingCostCents" : 0,
    "discountCents" : 0,
    "grandTotalCents" : 0,
    "totalQuantity" : 0,
    "purelyDigital" : false,
    "hasDigital" : false,
    "requiresShipping" : false
  },
  "shareButtons" : {
    "stumble" : true,
    "twitter" : true,
    "facebook" : true,
    "reddit" : true,
    "tumblr" : true,
    "pinterest" : true,
    "google" : true,
    "linkedin" : true
  },
  "showCart" : false,
  "empty" : false,
  "emptyFolder" : false,
  "calendarView" : false
}
```

**Comment by kitajchuk** [comment]
@skrichten - Did you try running `sqs buster` then restarting the server?

**Comment by skrichten** [comment]
yes

**Comment by skrichten** [comment]
If I had to guess, I'd say the local server is trying to render the folder instead of the first page in the folder.  I'm not sure that behavior is officially documented anywhere.  I only knew because I found it in the forums...
http://answers.squarespace.com/questions/303/how-do-i-give-a-folder-a-default-page

**Comment by kitajchuk** [comment]
@skrichten - This is true. The server actually checks for this specific scenario and handles it. Maybe there's more than one condition for it though.

https://github.com/NodeSquarespace/node-squarespace-server/blob/master/squarespace-server.js#L259

**Comment by kitajchuk** [comment]
@skrichten - The weird thing is when I do the same thing on a site I'm currently working on it works fine. I have a folder called "Test Folder". Inside there I have a page called "Test Folder Page". The url works just fine and correctly redirects to the first page in the folder.

**Comment by kitajchuk** [comment]
Is there something I'm missing about your case in particular?

**Comment by jasonbarone** [comment]
@skrichten: I'm wondering how the scope looks on your navigation. Somewhere in there, there needs to be a `{.section collection}`. Can you paste in your entire .block template?

**Comment by kitajchuk** [comment]
This is what the `menu-nav` markup looks like in my block partial.

```html
<nav class="menu-nav js-menu-nav screen" data-content-field="navigation-menu-nav">
    <div class="screen__wrap">
        <div class="menu-nav__col column--2of4">
            <a href="/" class="js-menu-nav-logo">
                <img class="js-logo" width="115" src="/assets/svg/instrument.svg" />
            </a>
        </div>
        <div class="menu-nav__col column--2of4 -text--right">
            {.repeated section items}
                {.folder?}
                <ul class="menu-nav__ul menu-nav__ul--{collection.urlId}">
                    {.repeated section items}
                        {.if collection.passwordProtected}
                            {# Ignore the private stuff #}
                        {.or}
                            <li class="menu-nav__li menu-nav__li--{collection.urlId}">
                                <a class="menu-nav__a menu-nav__a--{collection.urlId} js-menu-nav-item a-h {.if active}is-active{.end}" href="{collection.fullUrl}">{collection.navigationTitle}</a>
                            </li>
                        {.end}
                    {.end}
                </ul>
                {.end}
            {.end}
        </div>
    </div>
</nav>
```

We're specifically only using one `index` in our main navigation. But I added a `folder` as well and it worked as expected, redirecting to the first root page of the folder on the local server and on Squarespace.

**Comment by skrichten** [comment]
I'm using the Wells template from squarespace.  I only modified it so that the nav would link to the fullUrl even if the item is a folder...

```
<ul>
  {.repeated section items}

      <li class="{.section collection}{typeLabel}-collection{.end}{.section active} active-link{.end}{.folder?} folder {collection.urlId}-folder{.end}{.if folderActive} active-link active-folder{.end}{.section externalLink} external-link{.end}">

        {.folder?}

          {.section collection}
          <a href="{fullUrl}">{navigationTitle}</a>
          {.end}
          <div class="subnav">
            <ul>
              {.repeated section items}
                {.collection?}
                  <li class="{collection.typeLabel}-collection{.section active} active-link{.end}">
                    <a href="{collection.fullUrl}">{collection.navigationTitle}</a>
                  </li>
                {.end}
                {.section externalLink}
                  <li class="external-link">
                    <a href="{url}"{.section newWindow} target="_blank"{.end}>{title}</a>
                  </li>
                {.end}
              {.end}
            </ul>
          </div>

        {.or}

          {.section collection}
            <a href="{fullUrl}">{navigationTitle}</a>
          {.end}

          {.section externalLink}
            <a href="{url}"{.section newWindow} target="_blank"{.end}>{title}</a>
          {.end}


        {.end}

      </li>

  {.end}
</ul>
```

I can't imagine that it would make any difference, but my main navigation does have another folder (currently empty).  So the structure is this...

Folder 1 (empty)
Folder 2 (contains the "Contact Page" from the Wells template demo)

**Comment by kitajchuk** [comment]
Weird, when I visit your site I don't even see those links...

**Comment by kitajchuk** [comment]
Sometimes its hard to debug without looking at how things are setup in the CMS as well unfortunately. I'll do my best to help figure it out with though.

**Comment by skrichten** [comment]
The are labeled "Work" and "About".  "About" is the one that gives the error on the local server.  Work is an empty folder, but I dynamically add a second layer of naviation under it with DOM manipulation to get around the "no nested folders/indexes" limitation that squarespace has.

**Comment by kitajchuk** [comment]
Cool, I'll check again. You can totally nest folders/indexes. The layout looks a little wonk in the CMS, but if you check your actual `siteLayout` data, the relationships are correct. I just add `index` and `folder` to my `index.conf` to enable it. You can hit this path off your site to check the navigation data `api/commondata/GetSiteLayout/`.

```javascript
{
    "title": "Index",
    "folder": true,
    "index": true,
    "indexType": "stacked",
    "addText": "Add Things",
    "icon": "projects",
    "fullData": false,
    "acceptTypes": [
        "page",
        "album",
        "blog",
        "events",
        "gallery",
        "products",
        "collection",
        "listing",
        "index",
        "folder",
        "kit-of-parts"
    ]
}
```

**Comment by kitajchuk** [comment]
@skrichten - Good news. I recreated your nav structure on a fresh instance of the Wells template and got the error locally. Time to fix :)

**Comment by kitajchuk** [comment]
@skrichten - Just published the fix. Latest server version is now `+ node-squarespace-server@0.4.5`. This commit finds the cause of the error you were having 25e869541a3ace43645070d7d3b26f78f11140b5.

**Comment by skrichten** [comment]
Nice!

"You can totally nest folders/indexes."

I noticed that this worked with indexes, but I ran into trouble when using the navigation tags/templates.  Thanks for the tip about GetSiteLayout, that might prove useful.

**Comment by kitajchuk** [comment]
Sweet. Also, make sure you update to this version of the server: `+ node-squarespace-server@0.4.6` if you don't want the page json console logging on every request :/

Ma bad.

**Comment by mattnull** [comment]
@kitajchuk I am noticing that i get this same issue when my page makes a cross domain ajax request as well. I am using + node-squarespace-server@0.4.6

**Comment by mattnull** [comment]
@kitajchuk https://www.dropbox.com/s/9i7kibytnasqjt7/Screenshot%202015-10-05%2015.16.27.png?dl=0

**Comment by kitajchuk** [comment]
@mattnull - re-opened the issue. Can you provide any more context into your setup and the nature of the request? If you can hit that page direct with `?format=json` and paste it here that might help to figure it out as well.

You say you're using `jsonp` for the request? Any particular reason? Usually `jsonp` is used for CORS.

**Comment by mattnull** [comment]
I have a client app within my squarespace site. A few actually. They pull in data from a remote server. There actually isn’t anything in the json referencing this. It is a script tag that pulls in a JS app. That app pulls in data from a remote server.



——————————————————Matt Null / Human Design

CTO / Co-Founder




www.humandesign.agency

@iammattnull

**Comment by kitajchuk** [comment]
@mattnull - Ok. This could be tricky without any visibility. What is the URL of the request that is blowing up? The full url. Is it on or off site?

**Comment by mattnull** [comment]
It seems like the url wouldn’t matter. If I make any cross domain request sqs picks it up. I think the problem may be within the way that SQS grabs requests. Here is an example of how the request would look. Its standard JSONP. http://mydomain.com/v1/posts?type=twitter&limit=1&callback=mithril_callback_1444236149513_ma6emzij30





Cheers


——————————————————Matt Null / Human Design

CTO / Co-Founder




www.humandesign.agency

@iammattnull

**Comment by skrichten** [comment]
Help!!  I just started getting the error again, and I can't figure out what caused it.  Here is what I was doing when I first noticed it...  On my local I saw that reset.css was getting added to site.css as the last file, so I tried adding it to the config as the first item in the stylesheets list in template.conf.  Then I read on the sqs site that you are not supposed to have it in there so I removed it.  The next time I did sqs buster and reloaded the server, logged in, and I started getting this error.  I did a little debugging myself and found that the local server is requesting a URL like this for the home page....

```http://mydomain.sqs.com//```

Notice the trailing slash... I would expect this url to be just http://mydomain.sqs.com/ or http://mydomain.sqs.com/home/

Any ideas what is happening?

**Comment by kitajchuk** [comment]
@skrichten Do you have a trailing slash in your `siteurl` in your `template.conf`?

**Comment by skrichten** [comment]
No

**Comment by kitajchuk** [comment]
Looks like this snippet, which is in the middleware methods, is the culprit:
```js
url = (get( "siteurl" ) + "/" + url.replace( /^\/|\/$/g, "" ) + "/");
```

Needs to check and make sure that the `url.replace()` result is not an empty string. Otherwise, you end up with what you have now which is a double trailing slash.

**Comment by kitajchuk** [comment]
I'll open the issue on the middleware. Should be an easy fix :)

**Comment by skrichten** [comment]
Thanks... still not sure why this just started happening all of a sudden though.

**Comment by kitajchuk** [comment]
Yeah, seems like something that would have come up a long time ago. There was a recent merge into the middleware related to `siteurl` by @foleyatwork but I can't imagine it would have caused this issue. The merge simply ensures that your `siteurl` doesn't have a trailing slash when setting it in the config.

**Comment by skrichten** [comment]
Wondering if it might be something strange on my local computer.  I fixed the trailing slash issue and still got the error.  Then I tried some other sites that I haven't touched in awhile, and I get the error there as well.  I'm on windows 10 pro 64 bit.  Let me know if you have any other thoughts on what I can try / look at.

**Comment by kitajchuk** [comment]
Well, now I'm getting the same issue locally as well :(

**Comment by skrichten** [comment]
oh, snap!  I guess its not my computer then.  Did Squarespace change something on their end maybe?  Please keep me posted and let me know if/how I can help in any way.

**Comment by kitajchuk** [comment]
Ehyo, I just patched the middleware. Reinstall the server and make sure you get the latest version:

`+ node-squarespace-middleware@0.2.3`

**Comment by skrichten** [comment]
Back in action!  Thank You!

**Comment by foleyatwork** [comment]
@kitajchuk was this related to my change? If so, sorry.

**Comment by kitajchuk** [comment]
@foleyatwork - nope, wasn't related to your change. Not sure what triggered this to start happening, but it was definitely weird and needed an update to how the URLs are put together for middleware api methods.
